src/nlp/problems/hs13.jl

```diff
@@ -0,0 +1,199 @@
+export HS13
+
+"""
+    nlp = HS13()
+
+## Problem 13 in the Hock-Schittkowski suite
+
+```math
+\\begin{aligned}
+\\min \\quad & (x_1 - 2)^2 + x_2^2 \\\\
+\\text{s. to} \\quad & 0 \\geq (1 - x_1)^3 - x_2
+\\end{aligned}
+```
+
+Starting point: `[-2; -2]`.
+"""
+mutable struct HS13{T, S} <: AbstractNLPModel{T, S}
+  meta::NLPModelMeta{T, S}
+  counters::Counters
+end
+
+function HS13(::Type{T}) where {T}
+  meta = NLPModelMeta{T, Vector{T}}(
+    2,
+    ncon = 1,
+    x0 = T[-2; 2],
+    lcon = T[0],
+    ucon = T[Inf],
+    name = "HS13_manual",
+  )
+
+  return HS13(meta, Counters())
+end
+HS13() = HS13(Float64)
+
+function NLPModels.obj(nlp::HS13, x::AbstractVector)
+  @lencheck 2 x
+  increment!(nlp, :neval_obj)
+  return (x[1] - 2)^2 + x[2]^2
+end
+
+function NLPModels.grad!(nlp::HS13, x::AbstractVector{T}, gx::AbstractVector{T}) where {T}
+  @lencheck 2 x gx
+  increment!(nlp, :neval_grad)
+  gx .= T[2 * (x[1] - 2); 2 * x[2]]
+  return gx
+end
+
+function NLPModels.hess_structure!(nlp::HS13, rows::AbstractVector{Int}, cols::AbstractVector{Int})
+  @lencheck 3 rows cols
+  rows[1] = 1
+  rows[2] = 2
+  rows[3] = 2
+  cols[1] = 1
+  cols[2] = 1
+  cols[3] = 2
```

Comment:
You don't need `nnzh = 3`. Mixed derivatives are 0. `hess_coord` should be fixed as well.